--- a/dwl.c	2025-07-14 18:59:54.801153624 +0100
+++ b/dwl.c	2025-07-14 19:35:15.157568457 +0100
@@ -617,7 +617,14 @@
 		/* Change focus if the button was _pressed_ over a client */
 		xytonode(cursor->x, cursor->y, NULL, &c, NULL, NULL, NULL);
 		if (c && (!client_is_unmanaged(c) || client_wants_focus(c)))
+		{
+			// Check if the client is different from the currently focused one
+			// and then call focusclient for sloppy focus.
+			if (seat->keyboard_state.focused_surface != client_surface(c)) {
+				focusclient(c, 0); // Re-introduce the sloppy focus behavior
+			}
 			focusclient(c, 1);
+		}
 
 		keyboard = wlr_seat_get_keyboard(seat);
 		mods = keyboard ? wlr_keyboard_get_modifiers(keyboard) : 0;

