--- a/dwl.c	2025-07-14 18:59:54.801153624 +0100
+++ b/dwl.c	2025-07-14 19:35:15.157568457 +0100
@@ -616,8 +616,16 @@
 
 		/* Change focus if the button was _pressed_ over a client */
 		xytonode(cursor->x, cursor->y, NULL, &c, NULL, NULL, NULL);
-		if (c && (!client_is_unmanaged(c) || client_wants_focus(c)))
-			focusclient(c, 1);
+                if (c && (!client_is_unmanaged(c) || client_wants_focus(c)))
+                		{
+                			if (sloppyfocus) {
+				                if (c != selmon->sellt) { // Corrected from selmon->sel to selmon->sellt
+                					focusclient(c, 0);
+                				}
+                			} else {
+                				focusclient(c, 1);
+                			}
+                		}
 
 		keyboard = wlr_seat_get_keyboard(seat);
 		mods = keyboard ? wlr_keyboard_get_modifiers(keyboard) : 0;
